### PR TITLE
Detect lines fix

### PR DESCRIPTION
**Pull request by rcooke-ast**
This PR makes improvements to the previous detect_lines algorithm, and removes some cython code. The development suite passes. The test suite does not pass. I have fixed one of the bugs in the test suite. The remaining errors are not related to this PR (the problem is with `collate` in linetools, and the `test_npeaks` function in `test_objfind.py`).

Note, you will need to reinstall your PYPIT installation once this PR is merged to the master branch.

### CODE REVIEW

**Comment by rcooke-ast** [comment]
Note, at the same time, an arclines PR needs to be approved.

File: pypit/ararc.py [hunk 1]
```diff
@@ -96,15 +96,16 @@ def detect_lines(slf, det, msarc, censpec=None, MK_SATMASK=False):
...
+    mskcnt = np.sum(mask)
     while True:
         w = np.where(mask == 0)
         xfit = xrng[w]
         yfit = detns[w]
         ct = np.polyfit(xfit, yfit, bpfit)
         yrng = np.polyval(ct, xrng)
         sigmed = 1.4826*np.median(np.abs(detns[w]-yrng[w]))
-        w = np.where(detns > yrng+1.5*sigmed)
+        w = np.where(detns > yrng+1.0*sigmed)
```
Comment (profxj) [inline pypit/ararc.py]: should we make this a parameter?

**Comment by profxj** [comment]
I'll deal with the test failures in a separate PR

**Comment by rcooke-ast** [comment]
OK, implemented the suggested change. I'll merge when travis tests complete

**Pull request merged**